lesson10.md

```diff
@@ -0,0 +1,315 @@
+<!-- .slide: id="lesson10" -->
+
+# Basic Frontend - Fall 2020
+
+Lesson 10, Tuesday, 2020-09-29
+
+---
+
+### Recap: For vs. while
+
+* We use a `for` loop if we know in advance how often it will be executed.
+
+```js
+for (let i = 0; i < 1000; i++) {
+    console.log(i);
+}
+```
+
+* We use a `while` loop if we don't know how many times the loop will be executed.
+
+```js
+while (isHungry) {
+    eat();
+}
+```
+
+---
+
+### Refresher: Data Types
+
+```js
+// string
+let name = "Owen";
+
+// number
+let age = 28;
+
+// boolean
+let isProgrammer = true;
+
+// undefined
+let doNotKnow;
+```
+
+---
+
+<!-- .slide: id="lesson10:arrays" -->
+
+### Arrays
+
+An `array` is a container type that holds multiple values:
+
+```js
+// we create an empty array using []
+let emptyArray = [];
+
+// we put the values we want in square brackets
+// separated by commas
+let ages = [19, 33, 25, 40];
+let cities = ["London", "Paris", "Berlin"];
+```
+
+---
+
+Array can hold any type of value:
+
+```js
+let prices = [0.99, 1.49];
+```
+
+And any quantity:
+```js
+// I only have one favorite food
+let favoriteFoods = ["Pizza"];
+// An array holding 26 letters of the alphabet
+let alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l"]; //TODO
+```
+
+---
+
+### Analogy: Bookshelf
+
+A bookshelf represents an array of books
+
+![bookshelf](images/bookshelf.jpg)
+
+```js
+let books = ["Harry Potter", "The Hobbit", "Game of Thrones"];
+```
+
+---
+
+### Accessing elements
+
+We can access elements in the array by number using square brackets "`[]`"
+
+The numbering starts at `0` (think floors of a building):
```

Comment:
Not a good analogy. It's very different across countries what the ground floor is numbered at.
https://en.wikipedia.org/wiki/Storey#Numbering

I cannot think of a different analogy, though.

Reply:
@pektinasen How about "think floors of a German building"?